src/Gaffer/Random.cpp

```diff
@@ -298,22 +287,7 @@ unsigned long int Random::computeSeed( const Context *context ) const
 	std::string contextEntry = contextEntryPlug()->getValue();
 	if( contextEntry.size() )
 	{
-		const IECore::Data *contextData = nullptr;
-		try
-		{
-			contextData = context->get<IECore::Data>( contextEntry );
-		}
-		catch( ... )
-		{
-		}
-		if( contextData )
-		{
-			IECore::MurmurHash hash = contextData->Object::hash();
-			/// \todo It'd be nice if there was a way of getting the hash folded into an
-			/// int so we could avoid this jiggery pokery.
-			std::string s = hash.toString();
-			seed += boost::hash<std::string>()( s );
-		}
+		seed += context->variableHash( contextEntry ).h1();
```

Comment:
1) We should definitely be concerned, because currently this isn't consistent between runs, due to dependency on the entry name address.  I've added a performance test in 5cf8124, which shows that by using variableHash in hash(), I can cancel out the extra cost of a data alloc in compute(), and get performance basically the same as master, which I've done in 7369915
2) ContextTest::countContextHash32Collisions demonstrates that each 32 bit chunk of the MurmurHash appears to have fairly ideal statistical properties when hashing contiguous integers, which leaves me reasonably confident that MurmurHash's mixing is good enough that tacking a h1 ^ h2 on the end wouldn't improve it much.  If there were any simple operation that we could do that would improve the mixing - then the author of MurmurHash probably would be doing it already?